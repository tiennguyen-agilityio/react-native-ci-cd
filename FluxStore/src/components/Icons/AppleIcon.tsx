import {memo} from 'react';
import {TouchableOpacity} from 'react-native';
import Svg, {Path, Rect} from 'react-native-svg';

// Interfaces
import {IconProps} from '@/interfaces';

// Stores
import {useThemeStore} from '@/stores';

// Themes
import {colors} from '@/themes';

export const AppleIcon = memo(
  ({width = 42, height = 42, disabled = false, style, onPress}: IconProps) => {
    const {
      theme: {text},
    } = useThemeStore();

    return (
      <TouchableOpacity
        testID="apple-icon"
        activeOpacity={0.8}
        onPress={onPress}
        disabled={disabled}
        style={style}>
        <Svg width={width} height={height} viewBox="0 0 42 42" fill="none">
          <Path
            fillRule="evenodd"
            clipRule="evenodd"
            d="M24.19 13.513c.77-.93 1.287-2.225 1.146-3.513-1.107.045-2.447.738-3.24 1.667-.712.824-1.336 2.14-1.167 3.403 1.234.096 2.495-.627 3.263-1.557h-.002zm2.77 8.175c.032 3.329 2.921 4.437 2.952 4.452-.023.078-.46 1.578-1.521 3.128-.917 1.34-1.868 2.676-3.368 2.704-1.473.026-1.946-.874-3.631-.874-1.683 0-2.21.846-3.604.9-1.447.055-2.549-1.45-3.475-2.784-1.888-2.731-3.332-7.718-1.393-11.084.962-1.672 2.683-2.73 4.551-2.757 1.421-.028 2.762.956 3.63.956.87 0 2.5-1.183 4.212-1.01.718.03 2.731.29 4.025 2.183-.105.064-2.403 1.402-2.378 4.186"
            fill={text.primary}
          />
          <Path
            fillRule="evenodd"
            clipRule="evenodd"
            d="M24.19 13.513c.77-.93 1.287-2.225 1.146-3.513-1.107.045-2.447.738-3.24 1.667-.712.824-1.336 2.14-1.167 3.403 1.234.096 2.495-.627 3.263-1.557h-.002zm2.77 8.175c.032 3.329 2.921 4.437 2.952 4.452-.023.078-.46 1.578-1.521 3.128-.917 1.34-1.868 2.676-3.368 2.704-1.473.026-1.946-.874-3.631-.874-1.683 0-2.21.846-3.604.9-1.447.055-2.549-1.45-3.475-2.784-1.888-2.731-3.332-7.718-1.393-11.084.962-1.672 2.683-2.73 4.551-2.757 1.421-.028 2.762.956 3.63.956.87 0 2.5-1.183 4.212-1.01.718.03 2.731.29 4.025 2.183-.105.064-2.403 1.402-2.378 4.186"
            fill={text.primary}
          />
          <Rect
            opacity={0.14}
            x={0.5}
            y={0.5}
            width={41}
            height={41}
            rx={20.5}
            fill={text.light}
            stroke={colors.gray[700]}
          />
        </Svg>
      </TouchableOpacity>
    );
  },
);
