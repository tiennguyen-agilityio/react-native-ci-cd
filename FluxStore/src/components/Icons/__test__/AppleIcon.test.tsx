import {render, fireEvent, screen} from '@testing-library/react-native';

import {AppleIcon} from '../AppleIcon';

describe('AppleIcon', () => {
  const onPress = jest.fn();

  it('renders correctly (matches snapshot)', () => {
    const {toJSON} = render(<AppleIcon />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onPress when touched and not disabled', () => {
    render(<AppleIcon onPress={onPress} />);

    fireEvent.press(screen.getByTestId('apple-icon'));

    expect(onPress).toHaveBeenCalledTimes(1);
  });

  it('does NOT call onPress when disabled', () => {
    render(<AppleIcon onPress={onPress} disabled />);

    fireEvent.press(screen.getByTestId('apple-icon'));

    expect(onPress).not.toHaveBeenCalledTimes(0);
  });
});
