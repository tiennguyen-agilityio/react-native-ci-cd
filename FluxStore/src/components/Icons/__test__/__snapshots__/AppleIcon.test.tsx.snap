// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppleIcon renders correctly (matches snapshot) 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="apple-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={42}
    bbWidth={42}
    fill="none"
    focusable={false}
    height={42}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 42,
          "width": 42,
        },
      ]
    }
    vbHeight={42}
    vbWidth={42}
    width={42}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        clipRule={0}
        d="M24.19 13.513c.77-.93 1.287-2.225 1.146-3.513-1.107.045-2.447.738-3.24 1.667-.712.824-1.336 2.14-1.167 3.403 1.234.096 2.495-.627 3.263-1.557h-.002zm2.77 8.175c.032 3.329 2.921 4.437 2.952 4.452-.023.078-.46 1.578-1.521 3.128-.917 1.34-1.868 2.676-3.368 2.704-1.473.026-1.946-.874-3.631-.874-1.683 0-2.21.846-3.604.9-1.447.055-2.549-1.45-3.475-2.784-1.888-2.731-3.332-7.718-1.393-11.084.962-1.672 2.683-2.73 4.551-2.757 1.421-.028 2.762.956 3.63.956.87 0 2.5-1.183 4.212-1.01.718.03 2.731.29 4.025 2.183-.105.064-2.403 1.402-2.378 4.186"
        fill={
          {
            "payload": 4280098594,
            "type": 0,
          }
        }
        fillRule={0}
        propList={
          [
            "fill",
            "fillRule",
          ]
        }
      />
      <RNSVGPath
        clipRule={0}
        d="M24.19 13.513c.77-.93 1.287-2.225 1.146-3.513-1.107.045-2.447.738-3.24 1.667-.712.824-1.336 2.14-1.167 3.403 1.234.096 2.495-.627 3.263-1.557h-.002zm2.77 8.175c.032 3.329 2.921 4.437 2.952 4.452-.023.078-.46 1.578-1.521 3.128-.917 1.34-1.868 2.676-3.368 2.704-1.473.026-1.946-.874-3.631-.874-1.683 0-2.21.846-3.604.9-1.447.055-2.549-1.45-3.475-2.784-1.888-2.731-3.332-7.718-1.393-11.084.962-1.672 2.683-2.73 4.551-2.757 1.421-.028 2.762.956 3.63.956.87 0 2.5-1.183 4.212-1.01.718.03 2.731.29 4.025 2.183-.105.064-2.403 1.402-2.378 4.186"
        fill={
          {
            "payload": 4280098594,
            "type": 0,
          }
        }
        fillRule={0}
        propList={
          [
            "fill",
            "fillRule",
          ]
        }
      />
      <RNSVGRect
        fill={
          {
            "payload": 4294967295,
            "type": 0,
          }
        }
        height={41}
        opacity={0.14}
        propList={
          [
            "fill",
            "stroke",
          ]
        }
        rx={20.5}
        stroke={
          {
            "payload": 4286021264,
            "type": 0,
          }
        }
        width={41}
        x={0.5}
        y={0.5}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
